api.go

```diff
@@ -15,236 +15,285 @@
 package apidVerifyApiKey
 
 import (
-	"database/sql"
 	"encoding/json"
-	"fmt"
+	"errors"
+	"io"
+	"io/ioutil"
 	"net/http"
-	"net/url"
+	"strings"
 )
 
-type sucResponseDetail struct {
-	Key             string `json:"key"`
-	ExpiresAt       int64  `json:"expiresAt"`
-	IssuedAt        string `json:"issuedAt"`
-	Status          string `json:"status"`
-	Type            string `json:"cType"`
-	RedirectionURIs string `json:"redirectionURIs"`
-	AppId           string `json:"appId"`
-	AppName         string `json:"appName"`
+type apiManagerInterface interface {
+	InitAPI()
+	handleRequest(w http.ResponseWriter, r *http.Request)
+	verifyAPIKey(verifyApiKeyReq VerifyApiKeyRequest) (*VerifyApiKeySuccessResponse, *ErrorResponse)
 }
 
-type errResultDetail struct {
-	ErrorCode string `json:"errorCode"`
-	Reason    string `json:"reason"`
+type apiManager struct {
+	dbMan             dbManagerInterface
+	verifiersEndpoint string
+	apiInitialized    bool
 }
 
-type kmsResponseSuccess struct {
-	RspInfo sucResponseDetail `json:"result"`
-	Type    string            `json:"type"`
-}
-
-type kmsResponseFail struct {
-	ErrInfo errResultDetail `json:"result"`
-	Type    string          `json:"type"`
+func (a *apiManager) InitAPI() {
+	if a.apiInitialized {
+		return
+	}
+	services.API().HandleFunc(a.verifiersEndpoint, a.handleRequest).Methods("POST")
+	a.apiInitialized = true
+	log.Debug("API endpoints initialized")
 }
 
 // handle client API
-func handleRequest(w http.ResponseWriter, r *http.Request) {
+func (a *apiManager) handleRequest(w http.ResponseWriter, r *http.Request) {
 
-	db := getDB()
-	if db == nil {
-		w.WriteHeader(http.StatusServiceUnavailable)
-		w.Write([]byte("initializing"))
-		return
-	}
+	w.Header().Set("Content-Type", "application/json")
 
-	err := r.ParseForm()
-	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("Unable to parse form"))
-		return
-	}
+	var returnValue interface{}
+
+	if verifyApiKeyReq, err := validateRequest(r.Body, w); err == nil {
```

Comment:
if `err!=nil`, this will not write anything to ResponseWriter, and no log for it.